http/routing/route_collection.go

```diff
@@ -143,6 +144,8 @@ func (c *RouteCollection) Middleware(middleware ...inter.HttpMiddleware) inter.R
 		route.SetMiddleware(middleware)
 	}
 
+	c.globalMiddlewares = append(c.globalMiddlewares, middleware...)
```

Comment:
The only problem is that the `globalMiddlewares` is also filled with middlewares in other routes, but is not used there. Maybe not so bad. We can also make use of Fallback Routes. Then we put that on the api route group and on the web route group (in api.go or in route_service_provider.go). Then it is clear which middlewares are executed and which are not. In addition, you have more control over the difference between web and api. How do you think of that?

https://confetti-framework.github.io/docs/the-basics/routing.html#fallback-routes

Example:
```go
var Api = Group(
	Get("/ping", controllers.Ping).Name("ping"),
	Fallback(func(request inter.Request) inter.Response {
		return outcome.Json(map[string]string{"message": "not found"}).Status(404)
	}),
).
	Prefix("/api").
	Middleware(middleware.Api...).
	Name("api.")
```

Reply:
I’m not sure whether I fully understand the problem about middlewares from other routes. Do you mean that the name `globalMiddlewares` is inaccurate because for a route collection that is not bound as `routes` singleton on the app, the middlewares aren’t actually _global_ in the scope of the application? If so yes, that’s true. Maybe we could rename it to just `middlewares`.

---

I was trying to make a middleware that serves static files from the disk. I didn’t want to manually create a fallback route because I wanted to keep the [normal error behavior](https://confetti-framework.github.io/docs/the-basics/errors.html#custom-http-error-pages) of `encoder.ErrorToHtml` to fall back to the error page without writing any code for it.

Using a fallback route here sounds like an interesting idea. This would implicitly alleviate the issue that global middlewares don’t run when a route is not registered for the path. When a fallback route is present on the `Web` group, all global and route-specific middlewares are executed.

How would you roughly go about adding a standard fallback (if that’s what you’re suggesting :sweat_smile:) while retaining the standard not-found behavior?

Reply:
You totally convinced me. After some thinking, I don't think a fallback route is a very nice solution.

Actually now my only objection is the naming of `globalMiddlewares`. Renaming it to `middlewares` is indeed much better. Than we can use that for something else later too. If you change that, I'll merge it 😄

Reply:
Cool :smile: I’ve renamed the slice to `middlewares`, back to you! :ping_pong:

Reply:
Released in [v0.11.1](https://github.com/confetti-framework/foundation/releases/tag/v0.11.1)

I also pasted the HEAD together with the GET in that release (same as in Laravel)
![image](https://user-images.githubusercontent.com/16726304/153957015-e177253d-e297-4c58-b602-d4a3d17f0d99.png)